modules/rostests/apitests/shell32/SHLimitInputEdit.cpp

```diff
@@ -0,0 +1,332 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Test for SHLimitInputEdit
+ * PROGRAMMER:      Katayama Hirofumi MZ
+ */
+
+#include "shelltest.h"
+#include <shobjidl.h>
+#include <shlwapi.h>
+
+class CShellFolder
+    : public IShellFolder
+    , public IItemNameLimits
+{
+public:
+    CShellFolder(INT iMaxNameLen = 0, BOOL bDisabled = FALSE)
+        : m_nRefCount(1)
```

Comment:
Your reference counting is off, it should be initialized on 0.

Reply:
The reference count is important for this case. We have to investigate whether `AddRef` and `Release` are called.